src/duckling/project/project.service.ts

```diff
@@ -252,22 +255,60 @@ export class ProjectService {
      * @param metaDataFile The name of the meta data file stored in the .duckling/ folder (excluding file type)
      * @return the path that can be used to load the meta data file
      */
-    getUserMetaDataPath(metaDataFile : string) : string {
+    getUserMetaDataPath(metaDataFile: string): string {
         return this._pathService.join(this.userMetaDataDir, metaDataFile + ".json");
     }
 
-    private async _parseMapJson(json : any, key : string) {
-        let rawMap = json ? JSON.parse(json) : createRawMap(this._project.versionInfo.mapVersion);
+    async runExistingCodeMigration(migrationName: string, options: any): Promise<void> {
+        this._updateProjectVersionInfo(migrationName, options);
+        this._updateEntitySystemWithExistingCodeMigration(migrationName, options);
+    }
+
+    private _updateProjectVersionInfo(migrationName: string, options?: any) {
+        let newVersionInfo = this._migrationService.updateVersionInfoWithExistingCodeMigration(
+            this.project.value.versionInfo,
+            migrationName,
+            options);
+        this._storeService.dispatch(setVersionInfo(newVersionInfo));
+    }
+
+    private _updateEntitySystemWithExistingCodeMigration(migrationName: string, options: any) {
+        let newEntitySystem = this._migrationService.migrateEntitySystem(
+            this._entitySystem.entitySystem.value,
+            migrationName,
+            options);
+        this._storeService.dispatch(replaceSystemAction(newEntitySystem));
+    }
+
+    async openVersionFile(): Promise<any> {
```

Comment:
This function isn't used, and the functionality exists inside of project service